### PR TITLE
Scheduling Events logic

### LINKED ISSUE #81: As a social worker, I want to keep track of my important meetings/events with clients
UI element changes

### PR DESCRIPTION
**Pull request by Rishit02**
- Closes #81

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ScheduleDeleteCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        Heading heading = ParserUtil.parseHeading(argMultimap.getValue(PREFIX_HEADING).get());
+        return new ScheduleDeleteCommand(heading);
+    }
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
```
Comment (tejas-1405) [inline src/main/java/seedu/address/logic/parser/ScheduleDeleteCommandParser.java:41]: Good use of streams to check prefixes

File: src/main/java/seedu/address/logic/commands/ScheduleDeleteCommand.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import seedu.address.model.event.Event;
+import seedu.address.model.event.Heading;
+
+/**
+ * Deletes an event identified by its heading from the schedule.
+ */
+public class ScheduleDeleteCommand extends ScheduleCommand {
+
+    public static final String COMMAND_WORD = "delete";
+
```
Comment (tejas-1405) [inline src/main/java/seedu/address/logic/commands/ScheduleDeleteCommand.java:22]: can remove the extra lines in between the fields

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 3]
```diff
@@ -106,6 +106,8 @@ public class CommandTestUtil {
     public static final String INVALID_NOK_DESC = " " + PREFIX_NOK; // empty string not allowed for next of kin
     public static final String INVALID_UPDATE_DESC = " " + PREFIX_UPDATE + "James&"; // '&' not allowed in names
     public static final String INVALID_TAG_DESC = " " + PREFIX_TAG + "hubby*"; // '*' not allowed in tags
+    public static final String INVALID_TIME_DESC = " " + PREFIX_TIME + "Monday"; // Time must be in a spcecific format
```
Comment (tejas-1405) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java:109]: minor typo in comment

**Review by tejas-1405 — COMMENTED** [review]
Minor code quality changes

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java [hunk 4]
```diff
@@ -213,6 +214,26 @@ public String retrieveNextCommand() {
         public ReadOnlyCalendar getCalendar() {
             throw new AssertionError("This method should not be called.");
         }
+
+        @Override
+        public boolean hasEvent(Event event) {
```
Comment (eliotlee68) [inline src/test/java/seedu/address/logic/commands/AddCommandTest.java:219]: Good variable naming

File: src/main/java/seedu/address/model/event/Time.java [hunk 5]
```diff
@@ -1,29 +1,62 @@
...
     public static final DateTimeFormatter DATE_TIME_FORMATTER =
             DateTimeFormatter.ofPattern("EEEE, MMMM, dd, yyyy - hh:mm a");
-    private LocalDateTime localDateTime;
+    public final String time;
+    private final LocalDateTime localDateTime;
+
+    /**
+     * Constructs a Time object from a string representation of time.
+     *
+     * @param timeString The string representation of time.
```
Comment (eliotlee68) [inline src/main/java/seedu/address/model/event/Time.java:26]: string representation of *the* time.

File: src/test/java/seedu/address/logic/commands/ScheduleAddCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+    }
+
+    @Test
+    public void getCommandStringTest() {
+        ScheduleAddCommand scheduleAddCommand = new ScheduleAddCommand(MEETING_WITH_ALICE);
+        String expected = ScheduleAddCommand.COMMAND_WORD;
+        assertEquals(expected, scheduleAddCommand.getCommandString());
+    }
+    /**
+     * A default model stub that have all of the methods failing.
```
Comment (eliotlee68) [inline src/test/java/seedu/address/logic/commands/ScheduleAddCommandTest.java:100]: that *has* all of the methods failing

**Review by eliotlee68 — COMMENTED** [review]
LGTM other than very very minor grammar stuff

**Pull request merged**